consensus/polybft/block_builder.go

```diff
@@ -82,31 +61,37 @@ type BlockBuilder struct {
 	state *state.Transition
 }
 
-// Reset is used to indicate that the current block building has been interrupted
-// and it has to clean any data
+// Init initializes block builder before adding transactions and actual block building
 func (b *BlockBuilder) Reset() error {
+	// set the timestamp
+	parentTime := time.Unix(int64(b.params.Parent.Timestamp), 0)
+	headerTime := parentTime.Add(b.params.BlockTime)
+
+	if headerTime.Before(time.Now()) {
+		headerTime = time.Now()
+	}
+
 	b.header = &types.Header{
 		ParentHash:   b.params.Parent.Hash,
 		Number:       b.params.Parent.Number + 1,
 		Miner:        b.params.Coinbase[:],
 		Difficulty:   1,
-		ExtraData:    b.params.Extra,
 		StateRoot:    types.EmptyRootHash, // this avoids needing state for now
 		TxRoot:       types.EmptyRootHash,
 		ReceiptsRoot: types.EmptyRootHash, // this avoids needing state for now
 		Sha3Uncles:   types.EmptyUncleHash,
 		GasLimit:     b.params.GasLimit,
+		Timestamp:    uint64(headerTime.Unix()),
 	}
 
-	b.block = nil
-	b.txns = []*types.Transaction{}
```

Comment:
Is there any reason to move this variables?

Reply:
just because i wanted to group all `b.*` in same place